sb3_contrib/common/vec_env/wrappers/action_masker.py

```diff
@@ -0,0 +1,35 @@
+from typing import Callable, List
+
+import gym
+import numpy as np
+from stable_baselines3.common.vec_env import VecEnv, VecEnvWrapper
+from stable_baselines3.common.vec_env.base_vec_env import VecEnvObs, VecEnvStepReturn
+
+from sb3_contrib.common.wrappers import ActionMasker
+
+
+class VecActionMasker(VecEnvWrapper):
+    """
+    A wrapper for a VecEnv of envs that have themselves been wrapped with the ActionMasker wrapper.
+
+    :param venv: the VecEnv to wrap. The underlying envs must be wrapped with the ActionMasker wrapper.
+    """
+
+    def __init__(self, venv: VecEnv):
+        super().__init__(venv)
+
+        if not all(self.env_is_wrapped(ActionMasker)):
+            raise ValueError("VecEnv's envs must be wrapped with the ActionMasker wrapper")
+
+    def valid_actions(self) -> List[np.ndarray]:
+        return self.venv.env_method("valid_actions")
```

Comment:
is this wrapper really needed?
or is it just to make things look nicer?

because it looks like you could just define two helper functions, `valid_actions()` and `action_mask_fn()` without having to redefine a wrapper

Reply:
Yes, it's mostly syntactic sugar. We could make do with a helper around calling `venv.env_method("valid_actions")`.

The other use for this class is to indicate that masking is desired, but there's no need for that if it's an exception to use the algorithm without masking.

Reply:
>The other use for this class is to indicate that masking is desired, but there's no need for that if it's an exception to use the algorithm without masking.

if the user uses `PPO_MASK`, I hope that he/she knows that masking is required